### ISSUE TITLE
agent services are not created

### ISSUE THREAD

**Issue by gutschet** [jenkinsci/docker-swarm-plugin#36]
I used the docker-plugin before and am trying to migrate to docker-swarm-plugin so my build agents can use all the resources in the swarm. I configured the docker host uri (http://<my-manager-ip>:2376) and the jenkins url. After that, I was able to see my swarm nodes in the swarm dashboard so I think the besic configuration works.
Then I added an agent template in which I set lable "JSLAVE" and image "jenkins/slave". Because in our company network we need a proxy to access the internet, I set to ENV variables for http_proxy and https_proxy. Then I added a job that is bound to lable "JSLAVE" and tried a build.
I can see a new agent appear in the web ui, but it stays offline. The web ui shows the following error:
![swarm_plugin_error](https://user-images.githubusercontent.com/53481185/62134849-8200ff80-b2e1-11e9-8058-edd1f56e70fe.png)
As suggested in issue #26 I added a logger for org.jenkinsci.plugins.docker.swarm that says:

> Jul 30, 2019 1:08:05 PM FINE org.jenkinsci.plugins.docker.swarm.docker.api.request.ApiRequest
> JSON Request: http://10.253.246.4:2376/tasks?filters=%7B%22service%22%3A%5B%22agent-testjob_2_fgc-39177%22%5D%7D, {}
> Jul 30, 2019 1:08:05 PM FINE org.jenkinsci.plugins.docker.swarm.docker.api.request.ApiRequest
> API Request response success: [{"ID":"b6qbz2m7b9wckksmapmusdo8m","Version":{"Index":19579},"CreatedAt":"2019-07-30T13:07:52.314398699Z","UpdatedAt":"2019-07-30T13:08:02.508274347Z","Labels":{},"Spec":{"ContainerSpec":{"Image":"jenkins/slave","Env":["http_proxy=http://\<my-proxy\>.\<domain\>:\<port\>","https_proxy=http://\<my-proxy\>.\<domain\>:\<port\>","DOCKER_SWARM_PLUGIN_JENKINS_AGENT_SECRET=5c9ffea85f8bcf1dd5407366ff8182b7358f15ca2ff9fbdead8e880e996bda1d","DOCKER_SWARM_PLUGIN_JENKINS_AGENT_JAR_URL=https://jenkins.\<my-swarm-manager-ip\>.\<domain\>/jenkins-ci-fgc/jnlpJars/slave.jar","DOCKER_SWARM_PLUGIN_JENKINS_AGENT_JNLP_URL=https://jenkins.\<my-swarm-manager-ip\>.\<domain\>/jenkins-ci-fgc/computer/agent-testjob_2_fgc-39177/slave-agent.jnlp"],"Dir":"/tmp","User":"root","Isolation":"default"},"Resources":{"Limits":{},"Reservations":{}},"RestartPolicy":{"Condition":"none","MaxAttempts":0},"Placement":{"Constraints":["node.role==worker"]},"ForceUpdate":0},"ServiceID":"813cvuirq2u2erl8y1nrcqi96","Slot":1,"NodeID":"gvmhppulbd0gh2ffk0rq2drva","Status":{"Timestamp":"2019-07-30T13:08:02.219323615Z","State":"complete","Message":"finished","ContainerStatus":{"ContainerID":"08d6230ac6bfd7decd33efe678b091f049932f25923f60fdcff4cfb2cc537cf9","PID":0,"ExitCode":0},"PortStatus":{}},"DesiredState":"shutdown"}]
> Jul 30, 2019 1:08:05 PM INFO org.jenkinsci.plugins.docker.swarm.DeadAgentServiceReaperActor reapDeadAgentServices
> Reaping service: agent-testjob_2_fgc-39177
> Jul 30, 2019 1:08:05 PM FINE org.jenkinsci.plugins.docker.swarm.docker.api.request.ApiRequest
> JSON Request: http://10.253.246.4:2376/services/agent-testjob_2_fgc-39177, {}
> Jul 30, 2019 1:08:05 PM FINE org.jenkinsci.plugins.docker.swarm.docker.api.request.ApiRequest
> API Request response success
> Jul 30, 2019 1:13:05 PM FINE org.jenkinsci.plugins.docker.swarm.docker.api.request.ApiRequest
> JSON Request: http://10.253.246.4:2376/services?filters=%7B%22label%22%3A%5B%22ROLE%3Djenkins-agent%22%5D%7D, {}
> Jul 30, 2019 1:13:05 PM FINE org.jenkinsci.plugins.docker.swarm.docker.api.request.ApiRequest
> API Request response success: []

When I do docker service ls on my swarm manager host, I dont see a service being created.
In the plugins swarm dashboard, I can see the agent in the queue and it says "Provisioning: true", but it does not start.

**Comment by gutschet** [comment]
In the evening yesterday, I got one step further. I pulled the image jenkins/slave from docker hub, tagged it and pushed it to my local registry on the swarm. Now I can see a service being created with `docker service ls`. But as described in issue #26 it has replicas 0/1. In the swarm dashboard on the web ui the build stays in the queue, despite all cpus and 100% of memory are available. Wen I do `docker service logs -f <servicename>` it does not show any logs.
